### PR TITLE
cli: use generated clients in new-app and bootstrap

### PR DESCRIPTION
**Pull request by mfojtik**
@deads2k @bparees PTAL (only last commit)

This builds on top of https://github.com/openshift/origin/pull/16532 and will be rebased as soon as that PR will land.

### CODE REVIEW

File: pkg/generate/app/cmd/newapp.go [hunk 1]
```diff
@@ -121,7 +126,7 @@ type AppConfig struct {
 	CategoryExpander resource.CategoryExpander
 	ClientMapper     resource.ClientMapper
 
-	OSClient        client.Interface
+	// OSClient        client.Interface
```
Comment (bparees) [inline pkg/generate/app/cmd/newapp.go]: delete?

**Comment by mfojtik** [comment]
@bparees @deads2k we need some reasonable way to mock the Process() template otherwise those two tests I disabled for now will panic. I will spawn P1 issue to track that.

**Comment by mfojtik** [comment]
@deads2k fixing projects was a can of worms :-/ now half of the integration needs to be reworked...

**Comment by mfojtik** [comment]
@deads2k this picks your integration and adds my new-app/cluster up stuff... i will keep working on this today.

**Comment by mfojtik** [comment]
fire

**Comment by mfojtik** [comment]
@deads2k i plumbed authorization test, but got stuck in build admission:

```
 clients := map[string]*client.Client{"admin": projectAdminClient, "editor": projectEditorClient}
```

and there is more! :-) I will need to refact that test to be more sane.

File: pkg/image/generated/clientset/typed/image/v1/fake/fake_imagestreammapping.go [hunk 2]
```diff
@@ -2,6 +2,7 @@ package fake
 
 import (
 	v1 "github.com/openshift/origin/pkg/image/apis/image/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (coreydaley) [inline pkg/image/generated/clientset/typed/image/v1/fake/fake_imagestreammapping.go]: Why are we using meta_v1 in some places and metav1 in others for the same package? Seems confusing

Comment (mfojtik) [reply pkg/image/generated/clientset/typed/image/v1/fake/fake_imagestreammapping.go]: this is generated code

**Comment by deads2k** [comment]
I'll pick this up tomorrow.

**Pull request closed without merging**